mbed/mbed.py

```diff
@@ -1081,8 +1081,6 @@ def fromlib(cls, lib=None):
 
         if m_repo_ref:
             rev = m_repo_ref.group(3)
-            if rev and rev != 'latest' and rev != 'tip' and not re.match(r'^([a-fA-F0-9]{6,40})$', rev):
```

Comment:
We think this causes problems up stream when differentiating between SCMs. We can't put this in without further investigation.

Reply:
You mean the mercurial? I think people have to use the right types of .lib references for the tooling, it's once again something people need to solve, rather than the tool

Reply:
We're not preventing anyone from doing sensible things, even if we remove this limitation we have now.